contracts/Market.sol

```diff
@@ -126,6 +126,12 @@ contract Market is
    */
   IERC20WithPermit private _purchasingToken;
 
+  /**
+   * @notice The number of decimals configure on the _purchasingToken contract.
+   * Used to adapt between Removal amounts and payment token amounts.
+   */
+  uint8 private _purchasingTokenDecimals;
```

Comment:
there might be something valuable about the explicitness of storing this here and requiring it up on configuration of the purchasing token even if you were able to rely on calling `_purchasingToken.decimals()` to get that where necessary

Reply:
Totally, I'd rather do that but couldn't get it to work.  Would love it if someone can!

Reply:
Fixed, thanks!